jobbergate-api/jobbergate_api/s3_manager.py

```diff
@@ -40,32 +160,60 @@ def _get_key(self, app_id):
             )
         return self.key_template.format(app_id=app_id)
 
-    def put(self, upload_file: UploadFile, app_id: str = ""):
+    def put(self, upload_file: UploadFile, app_id: str = "") -> str:
         """
-        Upload a file to s3 for the given app_id and returns the.
+        Upload a file to s3 for the given app_id and returns its key.
         """
         key = self._get_key(app_id)
-        self.s3_client.put_object(
-            Body=upload_file.file,
-            Bucket=self.bucket_name,
-            Key=key,
-        )
+        logger.debug(f"Uploading to S3: {key})")
+        if isinstance(upload_file, str):
+            self.client.put(upload_file, key)
+        else:
+            self.client.put(upload_file.file, key)
```

Comment:
This block may be unnecessary if we use the S3 fixture properly. Investigate it.

Reply:
💯 

The type-hint says `upload_file` is an instance of `UploadFile`. Are we actually sending this method a string anywhere in our code-base?

Reply:
Yes, some tests send strings. I'm working on it.

Reply:
solved in [75b4341](https://github.com/omnivector-solutions/jobbergate/pull/111/commits/75b434114eb0bbfac0917da9c87101bb9c9eae79)